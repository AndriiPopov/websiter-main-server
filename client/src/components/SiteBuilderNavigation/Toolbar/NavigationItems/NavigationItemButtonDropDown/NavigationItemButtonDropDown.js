import React, { Component } from 'react';
import PropTypes from 'prop-types';

import classes from './NavigationItemButtonDropDown.module.css';
import Svg from '../../../../Svg/Svg';
import AUX from '../../../../../hoc/Aux';
import DropDownMenu from '../../../../../hoc/DropDownMenu/DropDownMenu';

export class NavigationItemButtonDropDown extends Component {
    state = {
        isVisible: false
    };

    handleOpen = () => {
        this.setState({ isVisible: !this.state.isVisible })
    };
    
    handleClose = () => {
        this.setState({ isVisible: false })
    };

    render () {
        return (
            <AUX>
                <li className={classes.NavigationItemButtonDropDown}
                    onClick={this.handleOpen}>
                    <Svg icon={this.props.icon} />
                </li>
                { this.state.isVisible ? 
                    <DropDownMenu 
                        close={this.handleClose}
                        isTopMenu
                        width={this.props.width} >
                        {this.props.children}
                    </DropDownMenu> : null 
                }
            </AUX>
        );
    };
};

NavigationItemButtonDropDown.propTypes = {
    icon: PropTypes.string,
    dropDownType: PropTypes.oneOf(['pageZoom', 'pagesMenu'])
};

export default NavigationItemButtonDropDown;