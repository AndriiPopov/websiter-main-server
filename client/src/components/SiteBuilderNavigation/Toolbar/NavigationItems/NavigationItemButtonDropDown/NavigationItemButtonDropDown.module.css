.NavigationItemButtonDropDown {
    color: #555;
    cursor: pointer;
    padding: 15px 10px;
    height: 100%;
    vertical-align: bottom;
    fill: #555;
    margin: 0px;
    display: flex;
    width: auto;
    align-items: center;
    box-sizing: border-box;
}

.NavigationItemButtonDropDown:hover,
.NavigationItemButtonDropDown:active {
    background: #eee;
}

.NavigationItemButtonDropDown a.disabled {
        color: grey;
}
