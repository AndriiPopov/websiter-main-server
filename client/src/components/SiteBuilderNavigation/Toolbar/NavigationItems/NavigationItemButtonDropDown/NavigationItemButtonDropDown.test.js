import React from 'react';
import {shallow, configure} from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import { NavigationItemButtonDropDown } from './NavigationItemButtonDropDown';

configure({adapter: new Adapter()});

describe('NavigationItemButtonDropDown', () => {
    it('should be defined', () => {
        expect(NavigationItemButtonDropDown).toBeDefined();
    });

    it('it should render button with onclick null when disabled', () => {
        const tree = shallow(
            <NavigationItemButtonDropDown
                disabled
                clicked="clicked"
                icon="<svg></svg>" />
        );
        expect(toJson(tree)).toMatchSnapshot();
    });

    it('it should render button with onclick', () => {
        const tree = shallow(
            <NavigationItemButtonDropDown
                clicked="clicked"
                icon="<svg></svg>" />
        );
        expect(toJson(tree)).toMatchSnapshot();
    });

    it('it should render a dropdown div with cover', () => {
        const tree = shallow(
            <NavigationItemButtonDropDown
                clicked="clicked"
                icon="<svg></svg>" />
        );
        tree.find('li').first().simulate('click');
        expect(toJson(tree)).toMatchSnapshot();
    });
});