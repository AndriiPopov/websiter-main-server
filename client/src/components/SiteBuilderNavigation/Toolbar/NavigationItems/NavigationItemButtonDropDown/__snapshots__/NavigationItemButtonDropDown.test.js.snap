// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationItemButtonDropDown it should render a dropdown div with cover 1`] = `
<aux>
  <li
    className="NavigationItemButtonDropDown"
    onClick={[Function]}
  >
    <Svg
      icon="<svg></svg>"
    />
  </li>
  <dropDownMenu
    close={[Function]}
  />
</aux>
`;

exports[`NavigationItemButtonDropDown it should render button with onclick 1`] = `
<aux>
  <li
    className="NavigationItemButtonDropDown"
    onClick={[Function]}
  >
    <Svg
      icon="<svg></svg>"
    />
  </li>
</aux>
`;

exports[`NavigationItemButtonDropDown it should render button with onclick null when disabled 1`] = `
<aux>
  <li
    className="NavigationItemButtonDropDown"
    onClick={[Function]}
  >
    <Svg
      icon="<svg></svg>"
    />
  </li>
</aux>
`;
